# Programs on if statement

**1. Write a program to perform addition of two numbers.**

In [2]:
num1 = 10
num2 = 5

addition = num1 + num2

print('Addition of {} and {} is'.format(num1,num2),addition)

Addition of 10 and 5 is 15


**2. Write a program to perform subtraction of two numbers.**

In [3]:
num1 = 10
num2 = 5

substraction = num1 - num2

print('Subtraction of {} and {} is'.format(num1,num2),substraction)

Subtraction of 10 and 5 is 5


**3. Write a program to perform multiplication of two numbers.**

In [4]:
num1 = 10
num2 = 5

multiplication = num1 * num2

print('Multiplication of {} and {} is'.format(num1,num2),multiplication)

Multiplication of 10 and 5 is 50


**4. Write a program to perform division of two numbers.**

In [5]:
num1 = 10
num2 = 5

division = num1 / num2

print('Division of {} and {} is'.format(num1,num2),division)

Division of 10 and 5 is 2.0


**5. Write a program to perform modulo operation.**

In [6]:
num1 = 10
num2 = 3

modulo = num1 % num2

print('Modulo of {} and {} is'.format(num1,num2),modulo)

Modulo of 10 and 3 is 1


**6. Write a program explaining ternary operator.**

In [7]:
number = 24

result = "Even" if number % 2 == 0 else "Odd"

print('{} is {}.'.format(number,result))

24 is Even.


**7. Write simple program using increment and decrement operator.**

In [8]:
a = 0
a += 1 #(a=a+1)
print("After incrementing a =", a)

a = 0
a -= 1 #(a=a-1)
print("After decrementing a =", a)

After incrementing a = 1
After decrementing a = -1


**8. Write a program to accept two integers and check whether they are equal or not.**

In [9]:
num1 = int(input('Enter first integer number : '))
num2 = int(input('Enter second integer number : '))

if num1 == num2:
    print('\n{} and {} is equal'.format(num1,num2))
else:
    print('\n{} and {} is not equal'.format(num1,num2))

Enter first integer number : 4
Enter second integer number : 3

4 and 3 is not equal


**9. Write a program to check whether a given number is positive or negative.**

In [10]:
number = -2

if number == 0:
    print('Number is zero')
else:
    if number > 0:
        print('{} is Positive'.format(number))
    else:
        print('{} is Negative'.format(number))

-2 is Negative


**10.Write a program to find whether a given year is a leap year or not.**

In [11]:
year = int(input('Enter year : '))

if year % 400 == 0 or year % 100 == 0 or year % 4 == 0:
    print('\n{} Year is Leap Year'.format(year))
else:
    print('\n{} Year is Not Leep Year'.format(year))

Enter year : 2024

2024 Year is Leap Year


**11. Write a program to calculate the root of a Quadratic Equation.**

In [12]:
from math import sqrt

while True:
    a = input('Enter a (Please enter only number...) : ')
    b = input('Enter b (Please enter only number...) : ')
    c = input('Enter c (Please enter only number...) : ')      
    if a.isdigit() and b.isdigit() and c.isdigit():
        a = int(a)
        b = int(b)
        c = int(c)
        print("\nYour Input number's is a = {} , b = {} , c = {}".format(a,b,c))
        print('Your Equation is : {}x^2 + {}x + {}'.format(a,b,c)) 
        if (b**2-4*a*c) > 0:
            Positive_alpha = (-b + sqrt(b**2-4*a*c))/(2*a)
            Negative_alpha = (-b - sqrt(b**2-4*a*c))/(2*a)
            print('\nAlpha1 is :',Positive_alpha)
            print('Alpha2 is :',Negative_alpha)
            exit = input('\nYou want to exit ? (yes/no) : ')
            if  exit == 'yes':
                print('Thank You !!')
                break
            else:
                pass
        else:
            print("\nQuadratic Equation Value will be nagative we can't find root of the negative value...")
            print('Please enter different values.')
            exit = input('\nYou want to exit ? (yes/no) : ')
            if  exit == 'yes':
                print('Thank You !!')
                break
            else:
                pass
    elif a.isalpha():
        print("\nEnter Valid Number's...")
        exit = input('\nYou want to exit ? (yes/no) : ')
        if  exit == 'yes':
            print('Thank You !!')
            break
        else:
            pass
    else:
        print("\nEnter Valid Number's...")
        exit = input('\nYou want to exit ? (yes/no) : ')
        if  exit == 'yes':
            print('Thank You !!')
            break
        else:
            pass

Enter a (Please enter only number...) : 1
Enter b (Please enter only number...) : 6
Enter c (Please enter only number...) : 8

Your Input number's is a = 1 , b = 6 , c = 8
Your Equation is : 1x^2 + 6x + 8

Alpha1 is : -2.0
Alpha2 is : -4.0

You want to exit ? (yes/no) : yes
Thank You !!


**12. Write a program to check whether a triangle can be formed by the given value for the angles.**

In [13]:
first_angle = float(input("Enter first angle: "))
second_angle = float(input("Enter second angle: "))
third_angle = float(input("Enter third angle: "))

if first_angle == 0 and second_angle == 0 and third_angle == 0:
    print('All zeros angle is not valid angle.')
else:
    total = first_angle + second_angle + third_angle
    print('\nYour Input 1st angle is {} degree , 2nd angle is {} degree , 3rd angle is {} degree'.format(first_angle,second_angle,third_angle))
    if total == 180:
        print("\nYes, a triangle can be formed with these angles.")
    else:
        print("\nNo, a triangle cannot be formed with these angles.")

Enter first angle: 90
Enter second angle: 0
Enter third angle: 90

Your Input 1st angle is 90.0 degree , 2nd angle is 0.0 degree , 3rd angle is 90.0 degree

Yes, a triangle can be formed with these angles.


**13. Write a program to check whether a character is an alphabet, digit or special character.**

In [14]:
special_char = ['!','@','#','$','%','^','&','*','/','?','<','>','<=','>=','"','-','+']

while True:
    character = input('\nCheck Your Input is Character , Alphabet or Digit ? (Enter "exit" to Quite...) : ')
    if character == 'exit':
        print('Thank You !!')
        break
    if character in special_char:
        print('{} is Special Character.'.format(character))
    elif character.isdigit():
        print('{} is Digit.'.format(character))
    elif character.isalpha():
        print('{} is Alphabet.'.format(character))
    else:
        print('Enter Valid Character...')


Check Your Input is Character , Alphabet or Digit ? (Enter "exit" to Quite...) : a
a is Alphabet.

Check Your Input is Character , Alphabet or Digit ? (Enter "exit" to Quite...) : 6
6 is Digit.

Check Your Input is Character , Alphabet or Digit ? (Enter "exit" to Quite...) : exit
Thank You !!


**14. Write a program to check whether an alphabet is a vowel or consonant.**

In [15]:
vowel = [97, 101, 105, 111, 117]
consonant = [98,99,100,102,103,104,106,107,108,109,110,112,113,114,115,116,118,119,120,121,122]

while True:
    new_char = input('\nCheck Your Input is Vowel or Consonant (Enter "exit" to Quite...): ')
    if new_char == 'exit':
        print('Thank you !!')
        break
    if len(new_char) > 1:
        print('Enter Valid Character...')
    elif ord(new_char.lower()) in vowel:
        print('{} is Vowel.'.format(new_char))
    elif ord(new_char.lower()) in consonant:
        print('{} is Consonant.'.format(new_char))
    else:
        print('Enter Valid Character...')


Check Your Input is Vowel or Consonant (Enter "exit" to Quite...): a
a is Vowel.

Check Your Input is Vowel or Consonant (Enter "exit" to Quite...): exit
Thank you !!


**15.Write a program to check whether a triangle is Equilateral, Isosceles or Scalene.**

In [16]:
side1 = float(input("Enter the length of side 1: "))
side2 = float(input("Enter the length of side 2: "))
side3 = float(input("Enter the length of side 3: "))

if side1 == side2 == side3:
    print("The triangle is Equilateral.")
elif side1 == side2 or side2 == side3 or side1 == side3:
    print("The triangle is Isosceles.")
else:
    print("The triangle is Scalene.")

Enter the length of side 1: 24
Enter the length of side 2: 34
Enter the length of side 3: 24
The triangle is Isosceles.


**16.Write a program to read any day number in integer and display day name in the word.**

In [17]:
day = int(input("Enter a day number : "))

if day == 1:
    print('Monday')
elif day == 2:
    print('Tuesday')
elif day == 3:
    print('Wednesday')
elif day == 4:
    print('Thursday')
elif day == 5:
    print('Friday')
elif day == 6:
    print('Saturday')
elif day == 7:
    print('Sunday')
else:
    print('Invalid day number')

Enter a day number : 3
Wednesday


**17.Write a program to accept a coordinate point in a XY coordinate system and determine in which quadrant the coordinate point lies.**

In [18]:
x = float(input("Enter the x-coordinate: "))
y = float(input("Enter the y-coordinate: "))

if x > 0 and y > 0:
    quadrant = "Quadrant I"
elif x < 0 and y > 0:
    quadrant = "Quadrant II"
elif x < 0 and y < 0:
    quadrant = "Quadrant III"
elif x > 0 and y < 0:
    quadrant = "Quadrant IV"
elif x == 0 and y != 0:
    quadrant = "On the y-axis"
elif x != 0 and y == 0:
    quadrant = "On the x-axis"
else:
    quadrant = "At the origin"

print(f"The point ({x}, {y}) lies in {quadrant}.")


Enter the x-coordinate: 2
Enter the y-coordinate: 0
The point (2.0, 0.0) lies in On the x-axis.


# Programs on loop

**18. Write a program to find the sum of first 10 natural numbers.**

In [19]:
total = 0
for i in range(1,11):
    total = total + i
    print('Number{} is : {}'.format(i,i))
print('\nSum of first 10 natural numbers is : ',total)

Number1 is : 1
Number2 is : 2
Number3 is : 3
Number4 is : 4
Number5 is : 5
Number6 is : 6
Number7 is : 7
Number8 is : 8
Number9 is : 9
Number10 is : 10

Sum of first 10 natural numbers is :  55


**19. Write a program to display n terms of natural number and their sum.**

In [20]:
n = input('Enter any number : ')

if n.isdigit():
    n = int(n)
    n_sum = 0
    for i in range(n+1):
        n_sum = n_sum + i
    #n_sum = n*(n+1) / 2
    print('\nSum of the n terms of natual number is : ',n_sum)
else:
    print('\nEnter Valid Number...')

Enter any number : 4

Sum of the n terms of natual number is :  10


**20. Write a program to display the cube of the number upto given an integer.**

In [21]:
n = 3

for i in range(n+1):
    cube = i**3
cube

27

**21. Write a program to display the multiplication table of a given integer.**

In [22]:
num = int(input('Enter a integer number : '))
for i in range(1,11):
    mul = num * i
    print('{} x {} = {}'.format(num,i,mul))

Enter a integer number : 4
4 x 1 = 4
4 x 2 = 8
4 x 3 = 12
4 x 4 = 16
4 x 5 = 20
4 x 6 = 24
4 x 7 = 28
4 x 8 = 32
4 x 9 = 36
4 x 10 = 40


**22. Write a program to calculate the factorial of a given number.**

In [23]:
num = int(input('Enter a integer number : '))
temp = 1
for i in range(1,num+1):
    temp = temp * i
print('\nFactorial of given number {} is {}'.format(num,temp))

Enter a integer number : 7

Factorial of given number 7 is 5040


**23. Write a program to display the n terms of even natural number and their sum.**

In [24]:
n = int(input("Enter the value of n: "))
count = 0
sum_even = 0
print("The first {} even natural numbers is :".format(n))
while count < n:
    even_number = 2 * count
    print(even_number, end=" ")
    sum_even = sum_even + even_number
    count += 1
print("\nThe sum of the first {} even natural numbers is: {}".format(n,sum_even))

Enter the value of n: 9
The first 9 even natural numbers is :
0 2 4 6 8 10 12 14 16 
The sum of the first 9 even natural numbers is: 72


**24. Write a program to display the sum of the series [ 9 + 99 + 999 + 9999 ...].**

In [25]:
num = int(input('Enter a integer number : '))
rem = 0
s = 0
print()
for i in range(1,num+1):
    rem  = rem * 10 + 9
    print(rem,'+ ',end='')
    s = s + rem
print('\n\nSum of the above series is : ',s)

Enter a integer number : 4

9 + 99 + 999 + 9999 + 

Sum of the above series is :  11106


**25. Write a program to check whether a given number is an armstrong number or not.**

In [26]:
num = int(input('Enter a integer number : ')) # 153 , 1634
temp = num
s = 0
length = len(str(num))
while num != 0:
    rem = num % 10
    s = s + (rem**length)
    num = num//10    
if s == temp:
    print('\nGiven number {} is Armstrong Number.'.format(temp))
else:
    print('\nGiven number {} is not Armstrong Number.'.format(temp))

Enter a integer number : 2

Given number 2 is Armstrong Number.


**26. Write a program to determine whether a given number is prime or not.**

In [27]:
number = 5

if number <= 1:
    print("Number is Not Prime...")
elif number == 2 or number == 3:
    print("Number is Prime...")
elif number % 2 == 0 or number % 3 == 0:
    print("Number is Not Prime...")
else:
    m = 5
    while m * m <= number:
        if number % m == 0 or number % (m + 2) == 0:
            print("Number is Not Prime...")
            break
        m = m + 6
    else:
        print("Number is Prime...")

Number is Prime...


**27. Write a program to display the first n terms of Fibonacci series.**

In [28]:
num = int(input('Enter a integer number : '))
num1, num2 = 0, 1
print('\nFibonacci Series : ')
print(num1,end=' ')
print(num2,end=' ')
for i in range(3,num+1):
    s = num1 + num2
    print(s,end=' ')
    num1 = num2
    num2 = s

Enter a integer number : 56

Fibonacci Series : 
0 1 1 2 3 5 8 13 21 34 55 89 144 233 377 610 987 1597 2584 4181 6765 10946 17711 28657 46368 75025 121393 196418 317811 514229 832040 1346269 2178309 3524578 5702887 9227465 14930352 24157817 39088169 63245986 102334155 165580141 267914296 433494437 701408733 1134903170 1836311903 2971215073 4807526976 7778742049 12586269025 20365011074 32951280099 53316291173 86267571272 139583862445 

**28. Write a program to display the number in reverse order.**
**Example: Input 12345 Output 54321**

In [29]:
num = int(input('Enter number : '))
s = 0
temp = num
while num != 0:
    rem = num % 10
    s = s * 10 + rem
    num = num//10
print('\nYour input number is : {}'.format(temp))
print('In Reverse Order : {}'.format(s))

Enter number : 54

Your input number is : 54
In Reverse Order : 45


**29. Write a program to check whether a number is a palindrome or not.**

In [30]:
num = int(input('Enter number : '))
s = 0
temp = num
while num != 0:
    rem = num % 10
    s = s * 10 + rem
    num = num//10
    
if temp == s:
    print('\n{} - Number is Palindrome'.format(s))
else:
    print('\n{} - Number is not Palindrome'.format(s))

Enter number : 43

34 - Number is not Palindrome


# Pattern programs

**30. Write a program to display the pattern like right angle triangle using an asterisk. The pattern like**

In [6]:
num = int(input('Enter One Number...'))
for i in range(num+1):
    for j in range(num+1):
        if (j>=1 and  j<=i):
            print('*',end='')
        else:
            print(' ',end='')
    print('\n')

Enter One Number...4
     

 *   

 **  

 *** 

 ****



**31.Write a program to display the pattern like right angle triangle with a number. The pattern like:**

In [7]:
n = int(input("Enter the number of rows: "))

for i in range(1, n + 1):
    for j in range(1, i + 1):
        print(j, end="")
    print()

Enter the number of rows: 5
1
12
123
1234
12345


**32. Write a program to make such a pattern like right angle triangle with a number which
will repeat a number in a row.**

In [9]:
num = int(input('Enter One Number...'))

for i in range(num+1):
    for j in range(num+1):
        if (j>=1 and  j<=i):
            print(i,end='')
        else:
            print(' ',end='')
    print('\n')

Enter One Number...4
     

 1   

 22  

 333 

 4444



**33.Write a program to make such a pattern like right angle triangle with number increased by 1. The pattern like:**

In [10]:
n = int(input("Enter the number of rows: "))
number = 1

for i in range(1, n + 1):
    for j in range(1, i + 1):
        print(number, end="")
        number = number + 1
    print()

Enter the number of rows: 4
1
23
456
78910


**34.Write a program to make such a pattern like a pyramid with numbers increased by 1. The pattern like:**

In [11]:
n = int(input("Enter the number of rows: "))
number = 1
for i in range(1, n + 1):
    for j in range(1, i + 1):
        print(number, end=" ")
        number += 1
    print()

Enter the number of rows: 6
1 
2 3 
4 5 6 
7 8 9 10 
11 12 13 14 15 
16 17 18 19 20 21 


**35. Write a program to print following pattern…**

In [13]:
num = int(input('Enter a number : '))
for i in range(num):
    for j in range(num+2):       
        if j>=1 and j <= 5:
            print('*'
                  ,end='')
        else:
            print(' ',end='')
    print()

Enter a number : 4
 *****
 *****
 *****
 *****


**36. . Write a program to print following pattern…**

In [14]:
n = int(input("Enter the number of rows: "))
for i in range(n):
    for j in range(n):
        if j >= i:
            if j == i or j == n - 1:
                print("*", end="")
            else:
                print(" ", end="")
        else:
            print(" ", end="")
    print()

Enter the number of rows: 5
*   *
 *  *
  * *
   **
    *


**37. Write a program to print following pattern…**

In [15]:
num = int(input('Enter a number : '))

for i in range(num):
    for j in range(num+1):
        if j>=1 and j<=num-i:
            print('*',end='')
        else:
            print(' ',end='')
    print()

Enter a number : 5
 *****
 **** 
 ***  
 **   
 *    


**38. Write a program to print following pattern…**

In [16]:
n = int(input("Enter the number of rows: "))

for i in range(n):
    for j in range(n):
        if i == 0 or i == n - 1 or j == 0 or j == i:
            print("*", end="")
        else:
            print(" ", end="")
    print()

Enter the number of rows: 5
*****
**   
* *  
*  * 
*****


**39. Write a program to print following pattern…**

In [17]:
n = int(input('Enter a number : '))
print()
for i in range(n+1):
    k=1
    for j in range((n*2)-1):
        if j>=i and j<=(n*2)-1-i and k:
            print('*',end='')
            k=0
        else:
            print(' ',end='')
            k=1
    print()

Enter a number : 4

* * * *
 * * * 
  * *  
   *   
       


**40. Write a program to print following pattern…**

In [18]:
n = int(input("Enter the number of rows: "))

for i in range(n):
    for j in range(n):
        if j >= i:
            print("*", end="")
        else:
            print(" ", end="")
    print() 

Enter the number of rows: 5
*****
 ****
  ***
   **
    *


**41. Write a program to print following pattern…**

In [19]:
n = int(input('Enter a number : '))
for i in range(1,n+1):
    for j in range(1,(n*2)):
        if j >= (n+1)-i and j <= (n-1)+i:
            print('*',end='')
        else:
            print(' ',end='')
    print()

Enter a number : 5
    *    
   ***   
  *****  
 ******* 
*********


**42. Write a program to print following pattern…**

**43. Write a program to print following pattern…**

**44. Write a program to print following pattern...**

In [20]:
n = int(input("Enter the number of rows: "))

for i in range(n):
    for j in range(i + 1):
        print("*", end="")
    print()

for i in range(n - 1):
    for j in range(n - i - 1):
        print("*", end="")
    print()

Enter the number of rows: 5
*
**
***
****
*****
****
***
**
*


**45. Write a program to read n number of values in an array and display it in reverse order.**

In [22]:
arr = []
n = int(input('Enter number of elements : '))
for i in range(0,n):
    num = int(input('element{}. : '.format(i+1)))
    arr.append(num)
print('\nInput Array : ',arr)
print('Array in reverse order : ',arr[::-1])

Enter number of elements : 5
element1. : 0
element2. : 1
element3. : -1
element4. : 4
element5. : -4

Input Array :  [0, 1, -1, 4, -4]
Array in reverse order :  [-4, 4, -1, 1, 0]


# Array programs

**46. Write a program to copy the elements of one array into another array.**

In [23]:
arr1 = [1,2,3,4,5]
arr2 = [11,12,13,14,15]
print('Array 1 : ',arr1)
print('Array 2 : ',arr2)
ele = int(input('\nEnter element of array 1 : '))
if ele in arr1:
    arr2.insert(0,ele)
    print('\nYour input element is : ',ele)
    print('Your element is copyed in the array 2 : ',arr2)
else:
    print('\nYour input {} is not present in array 1, Enter Valid element...'.format(ele))

Array 1 :  [1, 2, 3, 4, 5]
Array 2 :  [11, 12, 13, 14, 15]

Enter element of array 1 : 4

Your input element is :  4
Your element is copyed in the array 2 :  [4, 11, 12, 13, 14, 15]


 **47. Write a program to count a total number of duplicate elements in an array.**

In [24]:
l = [1,2,2,3,4,4,5]
new = []
c = 0
for i in l:
    if i in new:
        c = c + 1
    else:
        new.append(i)
print('Array is : ',l)
print('Total Number of Duplicate element is : ',c)

Array is :  [1, 2, 2, 3, 4, 4, 5]
Total Number of Duplicate element is :  2


**48. Write a program to print all unique elements in an array.**

In [25]:
arr = [1,2,2,3,4,4,4,5]
new_arr = []
for i in arr:
    if i not in new_arr:
        new_arr.append(i)
print('Array is : ',arr)
print('Unique element is : ',new_arr)

Array is :  [1, 2, 2, 3, 4, 4, 4, 5]
Unique element is :  [1, 2, 3, 4, 5]


**49. Write a program to merge two arrays of same size sorted in decending order.**

In [26]:
arr1 = [1,0,-5,2]
arr2 = [9,11,-2,16]

if len(arr1) == len(arr2):
    print('Array 1 is : ',arr1)
    print('Array 2 is : ',arr2)

    arr = arr1 + arr2
    print('Merge two array : ',arr)

    for i in range(len(arr)):
        for j in range(i + 1, len(arr)):
            if arr[i] < arr[j]:
                temp = arr[i]
                arr[i] = arr[j]
                arr[j] = temp

    print('Sorted in decending order : ',arr)
else:
    print('Array size is not same..')

Array 1 is :  [1, 0, -5, 2]
Array 2 is :  [9, 11, -2, 16]
Merge two array :  [1, 0, -5, 2, 9, 11, -2, 16]
Sorted in decending order :  [16, 11, 9, 2, 1, 0, -2, -5]


**50. Write a program to count the frequency of each element of an array.**

In [8]:
arr = [10,23,43,32,21,23,32,43,54,90,10]
frequency = {}
for ele in arr:
    if ele in frequency:
        frequency[ele] += 1
    else:
        frequency[ele] = 1
# print(frequency)
for key, value in frequency.items():
    print("Element {} occurs {} times".format(key,value))

Element 10 occurs 2 times
Element 23 occurs 2 times
Element 43 occurs 2 times
Element 32 occurs 2 times
Element 21 occurs 1 times
Element 54 occurs 1 times
Element 90 occurs 1 times


**51. Write a program to find the maximum and minimum element in an array.**

In [31]:
arr = [1,0,0,8,3,3,2,-5,-3,2]
print('Array is : ',arr)
for i in range(len(arr)):
    for j in range(i+1,len(arr)):
        if arr[i] < arr[j]:
            temp = arr[i]
            arr[i] = arr[j]
            arr[j] = temp
            
print('Maximum number is : ',arr[0])
print('Minimum number is : ',arr[9])

Array is :  [1, 0, 0, 8, 3, 3, 2, -5, -3, 2]
Maximum number is :  8
Minimum number is :  -5


**52. Write a program to separate odd and even integers in separate arrays.**

In [32]:
arr = [3,5,6,4,8,12,13,17,15]
even_arr = []
odd_arr = []
for i in arr:
    if (i % 2) == 0:
        even_arr.append(i)
    else:
        odd_arr.append(i)
print('Given Array is : {}'.format(arr))
print('Even elemtnts Array is : {}'.format(even_arr))
print('Odd elemtnts Array is : {}'.format(odd_arr))

Given Array is : [3, 5, 6, 4, 8, 12, 13, 17, 15]
Even elemtnts Array is : [6, 4, 8, 12]
Odd elemtnts Array is : [3, 5, 13, 17, 15]


**53. Write a program to sort elements of array in ascending order.**

In [33]:
arr = [1,7,3,5,0,9,2]
print('Input Array is : ',arr)
for i in range(len(arr)):
        for j in range(i + 1, len(arr)):
            if arr[i] > arr[j]:
                temp = arr[i]
                arr[i] = arr[j]
                arr[j] = temp
print('Array in ascending order : ',arr)

Input Array is :  [1, 7, 3, 5, 0, 9, 2]
Array in ascending order :  [0, 1, 2, 3, 5, 7, 9]


**54. Write a program to delete an element at desired position from an array.**

In [35]:
arr = [1,7,3,5,0,9,2]
print('Original Array is : ',arr)
num = int(input('Enter element : '))
for i in range(len(arr)):
    if num == arr[i]:
        index = i
print('\nDeleted element is : ',arr.pop(index))
print('Array is : ',arr)

Original Array is :  [1, 7, 3, 5, 0, 9, 2]
Enter element : 7

Deleted element is :  7
Array is :  [1, 3, 5, 0, 9, 2]


# Function Program

**55. Write a program to find the square of any number using the function**

In [36]:
def square(n):
    return n**2

num = int(input('Enter number for square : '))
print('\nSquare of {} is : {}'.format(num, square(num)))

Enter number for square : 4

Square of 4 is : 16


**56. Write a program to swap two numbers using function.**

In [37]:
def swap_number(a,b):
    b = a+b
    a = b-a
    b = b-a
    return a,b

a = int(input('Enter a : '))
b = int(input('Enter b : '))
print('\nOld a : ',a)
print('Old b : ',b)
a,b = swap_number(a,b)
print('\nNew a : ',a)
print('New b : ',b)

Enter a : 3
Enter b : 5

Old a :  3
Old b :  5

New a :  5
New b :  3


**57. Write a program to check a given number is even or odd using the function.**

In [38]:
def even_odd(n):
    if (n%2) == 0:
        return 'Even Number'
    else:
        return 'Odd Number'

num = int(input('Enter Number : '))
print('Your number {} is : {}'.format(num,even_odd(num)))

Enter Number : 4
Your number 4 is : Even Number


**58. Write a program to check whether a number is a prime number or not using the function.**

In [39]:
def prime_number(num):
    k = 1
    if num < 2:
        k = 0
    else:
        for i in range(2, num):
            if num % i == 0:
                k = 0
                break
    return k
num = int(input('Enter number to check its prime or not : '))
if prime_number(num):
    print("\n{} is a prime number".format(num))
else:
    print("\n{} is not a prime number".format(num))

Enter number to check its prime or not : 5

5 is a prime number


**59. Write a program to get the largest element of an array using the function.**

In [40]:
def find_largest():
    arr = [1,0,0,8,3,3,2,-5,-3,2]
    print('Array is : ',arr)
    for i in range(len(arr)):
        for j in range(i+1,len(arr)):
            if arr[i] < arr[j]:
                temp = arr[i]
                arr[i] = arr[j]
                arr[j] = temp
    return arr[0]
print('\nMaximum element in the array is : ',find_largest())

Array is :  [1, 0, 0, 8, 3, 3, 2, -5, -3, 2]

Maximum element in the array is :  8


**60. Write a program to check armstrong and perfect numbers using the function.**

In [41]:
def armstrong_number(num):
    temp = num
    s = 0
    length = len(str(num))
    while num != 0:
        rem = num % 10
        s = s + rem**length
        num = num // 10
    if temp == s:
        return 'Armstrong Number.'
    else:
        return 'Not Armstrong Number.'
def perfect(num):
    rem = 0
    for i in range(1, num):
        if num % i == 0:
            rem = rem + i
    if rem == num:
        print('{} is a Perfect number.'.format(num))
    else:
        print('{} is Not a Perfect number.'.format(num))
        
num = int(input('Enter number : '))
print('\n{} is {}'.format(num,armstrong_number(num)))
perfect(num)

Enter number : 5

5 is Armstrong Number.
5 is Not a Perfect number.


**61. Write a program to print all perfect numbers in given range using the function.**

In [43]:
low = int(input("Enter the lower limit of the range: "))
upr= int(input("Enter the upper limit of the range: "))

print("\nPerfect numbers in the range {} to {}:".format(low, upr))

for num in range(low, upr + 1):
    rem = 0
    for i in range(1, num):
        if num % i == 0:
            rem = rem + i

    if rem == num:
        print(num,end=" ")

Enter the lower limit of the range: 0
Enter the upper limit of the range: 500

Perfect numbers in the range 0 to 500:
0 6 28 496 

**62. Write a program to find out maximum and minimum of some values using function which will return an array.**

In [44]:
def max_min():
    arr = [1, 0, 8, 3, 9, 2, -5, -3, 2]
    print('Array is : ',arr)
    for i in range(len(arr)):
        for j in range(i+1,len(arr)):
            if arr[i] < arr[j]:
                temp = arr[i]
                arr[i] = arr[j]
                arr[j] = temp
    return arr
values = max_min()
print('Maximum Values is : ',values[0:3])
print('Minimum Values is : ',values[6:10])

Array is :  [1, 0, 8, 3, 9, 2, -5, -3, 2]
Maximum Values is :  [9, 8, 3]
Minimum Values is :  [0, -3, -5]


**63. Write a program to find the sum of the series 1!/1+2!/2+3!/3+4!/4+5!/5 using the function.**

In [45]:
def facto(n):
    temp = 1
    for i in range(1,n+1):
        temp = temp * i
    return temp

def series():
    s = 0
    for i in range(1,6):
        f = facto(i)
        s = s + f / i
    return s

print('Sum of the series is : ',series())

Sum of the series is :  34.0


**64. Write a program to convert decimal number to binary number using the function.**

In [46]:
def decimal_to_binary(n):
    s = []
    temp = n
    while n != 0:
        rem = n % 2
        s.append(rem)
        n = n//2
    s = s[::-1]
    b = 0
    for i in s:
        b = b * 10 + i
    return b,temp

n = int(input('Enter any number : '))
val,temp = decimal_to_binary(n)
print('\nDecimal Number : {} -> Binary Number : {}'.format(temp,val))

Enter any number : 7

Decimal Number : 7 -> Binary Number : 111


# Switch programs

**65. Write a program to read weekdays number and print weekday name using switch.**

In [48]:
switch = {
    1 : 'Monday.',
    2 : 'Tuesday.',
    3 : 'Wednesday.',
    4 : 'Thursday.',
    5 : 'Friday.',
    6 : 'Saturday.',
    7 : 'Sunday.'
}
day = int(input('Enter day number : '))
print("{} day is {}".format(day, switch.get(day)))

Enter day number : 5
5 day is Friday.


**66. Write a program to read gender and print corresponding gender using switch.**

In [11]:
gender = input("Enter gender (Male/Female): ")

gender_dict = {'Male': 'Male', 'Female': 'Female'}

if gender in gender_dict:
    print(f"The corresponding gender is: {gender_dict[gender]}")
else:
    print("Invalid gender code. Please enter 'Male' for Male or 'Female' for Female.")


Enter gender (Male/Female): Male
The corresponding gender is: Male


**67. Write a program to check whether a character is VOWEL or CONSONANT using switch.**

In [13]:
char = input("Enter a single character: ")
char = char.upper()
l = ['A', 'E', 'I', 'O', 'U']
if char.isalpha() and len(char) == 1:
    if char in l:
        print(f"The character '{char}' is a VOWEL.")
    else:
        print(f"The character '{char}' is a CONSONANT.")
else:
    print("Invalid input.")

Enter a single character: d
The character 'D' is a CONSONANT.


**68. Write a program to design calculator with basic operations using switch and also include features of scientific calculator.**

In [51]:
import math

print('Trigonomatric functions : ')
print('1. sin')
print('2. cos')
print('3. tan')
print('4. Square')
print('5. Cube')
print('6. Root')
print('7.Operations (+,-,/,*)')

number = int(input('\nEnter number (1,2,3,4,5,6,7,8)'))
if number == 1:
    x = int(input('\nEnter value of x : '))
    print('\nsin{} = {}'.format(x,math.sin(x)))
elif number == 2:
    x = int(input('\nEnter value of x : '))
    print('\ncos{} = {}'.format(x,math.cos(x)))
elif number == 3:
    x = int(input('\nEnter value of x : '))
    print('\ntan{} = {}'.format(x,math.tan(x)))
elif number == 4:
    x = int(input('\nEnter value of x : '))
    print('Square of {} is {}'.format(x,x**2))
elif number == 5:
    x = int(input('\nEnter value of x : '))
    print('Cube of {} is {}'.format(x,x**3))
elif number == 6:
    x = int(input('\nEnter value of x : '))
    print('Root of {} is {}'.format(x,math.sqrt(x)))
elif number == 7:
    num1 = int(input('Enter first number : '))
    op = input('Enter Operator (+,-,/,*,%) : ')
    num2 = int(input('Enter second number : '))
    if op == '+':
        print('{} + {} = {}'.format(num1,num2,num1+num2))
    elif op == '-':
        print('{} - {} = {}'.format(num1,num2,num1-num2))
    elif op == '*':
        print('{} x {} = {}'.format(num1,num2,num1*num2))
    elif op == '/':
        print('{} / {} = {}'.format(num1,num2,num1/num2))
    elif op == '%':
        print('{} % {} = {}'.format(num1,num2,num1%num2))
    else:
        print('Enter Valid Number...')
else:
    print('Enter Valid Number...')
    

Trigonomatric functions : 
1. sin
2. cos
3. tan
4. Square
5. Cube
6. Root
7.Operations (+,-,/,*)

Enter number (1,2,3,4,5,6,7,8)2

Enter value of x : 45

cos45 = 0.5253219888177297


**69. Write a program to check whether number is even or odd using switch.**

In [52]:
num = int(input('Enter number : '))

if num%2 == 0 :
    num = 1
else:
    num = 0
    
switch = {
    1 : '\nNumber is Even.',
    0 : '\nNumber is Odd.'
}
print(switch.get(num))

Enter number : 4

Number is Even.


**70. Write a program to find the number of days in month using switch case.**

In [53]:
switch = {
    'january' : '31 days',
    'february' : '28 or 29 days',
    'march' : '31 days',
    'april' : '30 days',
    'may' : '31 days',
    'june': '30 days',
    'july' : '31 days',
    'august' : '31 days',
    'september' : '30 days',
    'october' : '31 days',
    'november' : '30 days',
    'december' : '31 days'
}
month = input('Enter month name (january,february,...): ')
print("{} contain {}".format(month,switch.get(month,'Invalid')))

Enter month name (january,february,...): april
april contain 30 days


**71. Write a program to find the maximum between two numbers using switch case.**

In [54]:
num1 = int(input('Enter first number : '))
num2 = int(input('Enter second number : '))

if num1 > num2 :
    num1 = 1
else:
    num1 = 0
    
switch = {
    1 : '\nNumber1 is larger then the number2.',
    0 : '\nNumber2 is larger then the number1.'
}
print(switch.get(num1))

Enter first number : 5
Enter second number : 3

Number1 is larger then the number2.


**72. Write a program to check whether the number is positive, negative or zero using
switch case.**

In [55]:
num = int(input('Enter number : '))
if num == 0:
    n = 0
elif num < 0:
    n = 1
else:
    n = 2
switch = {
    0 : '\nNumber is Zero.',
    1 : '\nNumber is Negative.',
    2 : '\nNumber is Positive.'
}
print(switch.get(n))

Enter number : 4

Number is Positive.


**73. Write a program to find roots of a quadratic equation using switch case.**

# Basic programs

**74. Write a program to convert specified days into years, weeks and days.**

In [56]:
days = int(input('Enter Days : '))

weeks = days / 7
month = days / 30
year = days / 365
print()
print(weeks,'Weeks')
print(month,'Months')
print(year,'Year')

Enter Days : 56

8.0 Weeks
1.8666666666666667 Months
0.15342465753424658 Year


**75. Write a program that accepts an employee's ID, total worked hours of a month and
the amount he received per hour. Print the employee's ID and salary (with two
decimal places) of a particular month.**

In [57]:
e_id = input("Enter employee's ID : ")
e_hours = float(input('Enter total worked hours of a month : '))
e_sal = float(input('Enter amount he received per hour : '))

total_sal = e_hours * e_sal


print("\nEmployee's ID is {} Total salary of a month is : {}".format(e_id, total_sal))

Enter employee's ID : 101
Enter total worked hours of a month : 270
Enter amount he received per hour : 300

Employee's ID is 101 Total salary of a month is : 81000.0


**76. Write a program to calculate the distance between the two points.**

In [58]:
pointA = float(input('Enter Point A : '))
pointB = float(input('Enter Point B : '))

if pointA > 0 and pointB > 0:
    diff = pointB - pointA
    print(diff)
elif pointA < 0 and pointB < 0:
    diff = pointB - pointA
    print(diff)
elif pointA < 0 and pointB > 0:
    diff = pointB - pointA
    print(diff)
elif pointA > 0 and pointB < 0:
    diff = pointB - pointA
    print(diff)
elif pointA == 0 and pointB == 0:
    print('Distance is zero.')
elif pointA == 0 and pointB < 0:
    diff  = pointB - pointA
    print(diff)
elif pointA == 0 and pointB > 0:
    diff  = pointB - pointA
    print(diff)
elif pointA < 0 and pointB == 0:
    diff  = pointB - pointA
    print(diff)
elif pointA > 0 and pointB == 0:
    diff  = pointB - pointA
    print(diff)

Enter Point A : 3
Enter Point B : 0
-3.0


**77. Write a program to read an amount (integer value) and break the amount into smallest possible number of bank notes.**

In [27]:
amount = int(input("Enter the amount: "))

l = [500, 100, 50, 20, 10, 5, 2, 1]

note_count = {}

for i in l:
    count = amount // i
    if count > 0:
        note_count[i] = count
        amount = amount % i

for i, count in note_count.items():
    print("{} rupee : {} notes".format(i,count))

Enter the amount: 676
500 rupee : 1 notes
100 rupee : 1 notes
50 rupee : 1 notes
20 rupee : 1 notes
5 rupee : 1 notes
1 rupee : 1 notes


**78. Write a program to convert a given integer (in seconds) to hours, minutes and seconds.**

In [59]:
second = int(input('Enter number : '))

minutes = second/60
hours = second/3600
print('{} second = {} minutes'.format(second,minutes))
print('{} second = {} hours'.format(second,hours))

Enter number : 40
40 second = 0.6666666666666666 minutes
40 second = 0.011111111111111112 hours


**79. Write a program to convert a given integer (in days) to years, months and days,assumes that all months have 30 days and all years have 365 days.**

In [60]:
days = int(input('Enter Days : '))

weeks = days / 7
month = days / 30
year = days / 365
print()
print(weeks,'Weeks')
print(month,'Months')
print(year,'Year')

Enter Days : 4

0.5714285714285714 Weeks
0.13333333333333333 Months
0.010958904109589041 Year


**80. Write a program that accepts 4 integers p, q, r, s from the user where r and s are
positive and p is even. If q is greater than r and s is greater than p and if the sum of r
and s is greater than the sum of p and q print "Correct values", otherwise print
"Wrong values".**

In [61]:
for p in range(0,5):
    for q in range(0,5):
        for r in range(0,5):
            for s in range(0,5):
                if p%2 == 0 and r > 0 and s>0 and q > r and q > s and q > p and r+s > p+q:
                    print(p,q,r,s)

0 3 2 2
0 4 2 3
0 4 3 2
0 4 3 3


In [63]:
p = int(input('Enter p : '))
q = int(input('Enter q : '))
r = int(input('Enter r : '))
s = int(input('Enter s : '))

if p%2 == 0 and r > 0 and s>0 and q > r and q > s and q > p and r+s > p+q:
    print('\nCorrect values')
else:
    print('\nWrong values')

Enter p : 0
Enter q : 4
Enter r : 3
Enter s : 3

Correct values


**81. Write a program that read 5 numbers and sum of all odd values between them.**

In [64]:
l = []
print('Enter 5 numbers')
for i in range(1,6):
    n = int(input())
    l.append(n)
print('Sum of all odd number in 5 numbers is : ',sum(list(filter(lambda x : x if x%2!=0 else '',l))))

Enter 5 numbers
4
2
5
7
1
Sum of all odd number in 5 numbers is :  13


**82. Write a program that prints all even numbers between 1 and 50 (inclusive).**

In [65]:
for i in range(1,51):
    if i%2==0:
        print(i)

2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48
50


**83. Write a program that read 5 numbers and counts the number of positive numbers
and negative numbers.**

In [67]:
l = []
print('Enter 5 numbers : ')
for i in range(1,6):
    n = int(input())
    l.append(n)
print('Negative Number is : ',len(list(filter(lambda x : x if x<0 else '',l))))
print('Positive Number is : ',len(list(filter(lambda x : x if x>0 else '',l))))

Enter 5 numbers : 
6
5
4
3
2
Negative Number is :  0
Positive Number is :  5


**84. Write a program that read 5 numbers and counts the number of positive numbers
and print the average of all positive values.**

In [68]:
l = []
print('Enter Only 5 numbers : ')
for i in range(1,6):
    n = int(input())
    l.append(n)

postive_number = list(filter(lambda x : x if x>0 else '',l))
print('Positive Numbers is : ',postive_number)
print('Average of Positive Number is : ',sum(postive_number)/len(postive_number))

Enter Only 5 numbers : 
1
7
5
4
3
Positive Numbers is :  [1, 7, 5, 4, 3]
Average of Positive Number is :  4.0


**85. Write a program to find and print the square of each one of the even values from 1 to
a specified value.**

In [70]:
num = int(input('Enter number for (1 to a specified value.) : '))
even = []
for i in range(1,num+1):
    if i%2==0:
        even.append(i)
sqr = list(map(lambda x :x*2,even))
print('Even numbers is : ',even)
print('The square of each one of the even values : ',sqr)

Enter number for (1 to a specified value.) : 10
Even numbers is :  [2, 4, 6, 8, 10]
The square of each one of the even values :  [4, 8, 12, 16, 20]


**86. Write a program to check whether a given integer is positive even, negative even,
positive odd or negative odd. Print even if the number is 0.**

In [71]:
num = int(input('Enter number : '))

a = lambda x : (lambda y : 'Positive even' if y%2==0 else 'Positive Odd') if x > 0 else (lambda y : 'Negative even' if y%2==0 else 'Negative Odd')
b = a(num)
b(num)

Enter number : 5


'Positive Odd'

**87. Write a program to print all numbers between 1 to 100 which divided by a specified
number and the remainder will be 3.**

In [21]:
num = 7
for i in range(1,101):
    if i % 7 == 3:
        print(i)

3
10
17
24
31
38
45
52
59
66
73
80
87
94


**88. Write a program that accepts some integers from the user and find the highest value
and the input position.**

In [73]:
new = []
print('Enter 5 numbers : ')
for i in range(5):
    num = int(input())
    new.append(num)
print('Highest Value : ',max(new))
print('Input Position : ',id(max(new)))

Enter 5 numbers : 
7
5
4
3
2
Highest Value :  7
Input Position :  8885512


**89. Write a program to read a password until it is correct. For wrong password print
"Incorrect password" and for correct password print "Correct password" and quit the
program. The correct password is 1234.**

In [75]:
password = int(input('Enter Password : '))

if password == 1234:
    print('Correct password')
else:
    print('Incorrect password')

Enter Password : 1234
Correct password


**90. Write a program to read the coordinates(x, y) (in Cartesian system) and find the
quadrant to which it belongs (Quadrant -I, Quadrant -II, Quadrant -III, Quadrant -IV).**

In [76]:
x = float(input('Enter value of x : '))
y = float(input('Enter value of y : '))

if x > 0 and y > 0:
    print('{} and {} present in Quadrant - I'.format(x,y))
elif x < 0 and y > 0:
    print('{} and {} present in Quadrant - II'.format(x,y))
elif x < 0 and y < 0:
    print('{} and {} present in Quadrant - III'.format(x,y))
elif x > 0 and y < 0:
    print('{} and {} present in Quadrant - IV'.format(x,y))
elif x == 0 and y == 0:
    print('Origin')
elif x == 0 and y > 0:
    print('{} and {} present in Y-axis'.format(x,y))
elif y == 0 and x > 0:
    print('{} and {} present in X-axis'.format(x,y))
    
elif x == 0 and y < 0:
    print('{} and {} present in Negative Y-axis'.format(x,y))
elif y == 0 and x < 0:
    print('{} and {} present in Negative X-axis'.format(x,y))
else:
    print('Enter Valid NUmber...')

Enter value of x : 5
Enter value of y : 0
5.0 and 0.0 present in X-axis


**91. Write a program that reads two numbers and divide the first number by second
number. If the division not possible print "Division not possible".**

In [77]:
num1 = float(input('Enter first number : '))
num2  = float(input('Enter second number : '))

if  num2 == 0:
    print('Division not possible')
else:
    print('Division possible')

Enter first number : 4
Enter second number : 3
Division possible


**92. Write a program to find all numbers which are dividing by 7 and the remainder is
equal to 2 or 3 between two given integer numbers.**

In [78]:
for i in range(100):
    rem = i/7
    if rem>=2 and rem<=3:
        print(i)

14
15
16
17
18
19
20
21


**93. Write a program to print 3 numbers in a line, starting from 1 and print n lines. Accept
number of lines (n, integer) from the user.**

In [31]:
n = int(input("Enter the number of lines: "))

number = 1

for i in range(1, n + 1):
    for j in range(3):
        print(number, end=" ")
        number = number + 1
    print()

Enter the number of lines: 4
1 2 3 
4 5 6 
7 8 9 
10 11 12 


# Pointer programs

# String Programs

**110. Write a program to input a string and print it.**

In [31]:
input_string = input('Input String : ')
print('Input String is : ',input_string)

Input String : AJay
Input String is :  AJay


**111. Write a program to find the length of a string without using library function.**

In [81]:
string = 'Ajay Moladiya'
c=0
for i in string:
    c = c+1
    
print("Length of '{}' is : {}".format(string,c))

Length of 'Ajay Moladiya' is : 13


**112. Write a program to separate the individual characters from a string.**

In [82]:
string = 'Ajay Moladiya'

for i in string:
    print(i)

A
j
a
y
 
M
o
l
a
d
i
y
a


**113. Write a program to print individual characters of string in reverse order.**

In [83]:
string = 'Ajay Moladiya'
l = []
for i in string:
    l.append(i)
print('Characters of string in reverse order : ',l[::-1])

Characters of string in reverse order :  ['a', 'y', 'i', 'd', 'a', 'l', 'o', 'M', ' ', 'y', 'a', 'j', 'A']


**114. Write a program to compare two string without using string library functions.**

In [84]:
string1 = 'Ajaym'
string2 = 'Ajay'

if string1 == string2:
    print("'{}' and '{}' is Same".format(string1, string2))
else:
    print("'{}' and '{}' Not Same".format(string1, string2))

'Ajaym' and 'Ajay' Not Same


**116. Write a program to sort a string array in ascending order.**

In [85]:
import numpy as np

s_arr = np.array(['Ajay','Monark','Prashant','Keval'])

s_arr.sort()

In [86]:
s_arr

array(['Ajay', 'Keval', 'Monark', 'Prashant'], dtype='<U8')

**117. Write a program to demonstrate strcat**

**118. Write a program to demonstrate the working of strrchr()**

**119. Write a program to illustrate strcpy() function.**

# Class and object programs

**120. Create a class named 'Student' with a string variable 'name' and an integer variable
'roll_no'. Assign the value of roll_no as '2' and that of name as "xyz" by creating an
object of the class Student.**

In [87]:
class Student:
    name = 'xyz'
    roll_no = 2
s = Student()

In [88]:
s.name , s.roll_no

('xyz', 2)

**121. Assign and print the roll number, phone number and address of two students
having names "xyz" and "abc" respectively by creating two objects of the class
'Student'.**

In [89]:
class Student():
    name = 'xyz'
    roll_no = 2
    phone_no = 9867785678
    
    name1 = 'abc'
    roll_no1 = 1
    phone_no1 = 9078897889
a = Student()
b = Student()

In [90]:
a.name

'xyz'

In [91]:
a.name1

'abc'

In [92]:
b.name

'xyz'

**122. Write a program to print the area and perimeter of a triangle having sides of 3, 4
and 5 units by creating a class named 'Triangle' with a function to print the area and
perimeter.**

In [93]:
import math
class Triangle:
    def area(self,a,b,c):
        s = (a + b + c) / 2
        a = math.sqrt(s * (s - a) * (s - b) * (s - c))
        return a
    def perimeter(self,a,b,c):
        peri = a + b + c
        return peri

obj = Triangle()
area = obj.area(3, 4, 5)
perimeter = obj.perimeter(3,4,5)

print(f"Area of the triangle : ",area)
print(f"Perimeter of the triangle :",perimeter)

Area of the triangle :  6.0
Perimeter of the triangle : 12


**123.Write a program to print the area and perimeter of a triangle having sides of 3, 4
and 5 units by creating a class named 'Triangle' with the constructor having the
three sides as its parameters.**

In [94]:
import math
class Triangle:
    def __init__(self, a, b, c):
        self.a = a
        self.b = b
        self.c = c
    def area(self):
        s = (self.a + self.b + self.c) / 2
        a = math.sqrt(s * (s - self.a) * (s - self.b) * (s - self.c))
        return a
    def perimeter(self):
        peri = self.a + self.b + self.c
        return peri
    def print_f(self):
        area = self.area()
        perimeter = self.perimeter()
        print(f"Area of the triangle : ",area)
        print(f"Perimeter of the triangle :",perimeter)

obj = Triangle(3, 4, 5)
obj.print_f()

Area of the triangle :  6.0
Perimeter of the triangle : 12


**124. Write a program to print the area of two rectangles having sides (4,5) and (5,8)
respectively by creating a class named 'Rectangle' with a function named 'Area'
which returns the area. Length and breadth are passed as parameters to its
constructor.**

In [95]:
class Rectangle:
    def __init__(self,l,b):
        self.l = l
        self.b = b
    def Area(self):
        a = self.b*self.l
        return a
    
r1 = Rectangle(4,5)
r2 = Rectangle(5,8)
print('Area of Ractangle 1 is :',r1.Area())
print('Area of Ractangle 2 is : ',r2.Area())

Area of Ractangle 1 is : 20
Area of Ractangle 2 is :  40


**125. Print the average of three numbers entered by the user by creating a class named
'Average' having a function to calculate and print the average without creating any
object of the Average class.**

In [1]:
class Average:
    def calculate_and_print_average():
        num1 = float(input("Enter the first number: "))
        num2 = float(input("Enter the second number: "))
        num3 = float(input("Enter the third number: "))
        avg = (num1 + num2 + num3) / 3
        print("\nAverage:", avg)

Average.calculate_and_print_average()

Enter the first number: 5
Enter the second number: 3
Enter the third number: 9

Average: 5.666666666666667


**126. Write a program that would print the information (name, year of joining, salary,
address) of three employees by creating a class named 'Employee'. The output
should be as follows: Name Year of joining Address xyz 2000 Facebook tower abc
2015 google stree**

In [2]:
class Employee:
    name = input("Enter the name : ")
    year = int(input("Enter the joining year : "))
    join = input("Enter Address: ")
    
    print('\n{} {} {}'.format(name,year,join))

Enter the name : Ajay
Enter the joining year : 2023
Enter Address: Petpooja

Ajay 2023 Petpooja


**127. Write a program by creating an 'Employee' class having the following functions and
print the final salary.
1 - 'getInfo()' which takes the salary, number of hours of work per day of employee
as parameters
2 - 'AddSal()' which adds 1000 to the salary of the employee if it is less than 5000.
3 - 'AddWork()' which adds 500 to the salary of the employee if the number of
hours of work per day is more than 6 hours.**

In [3]:
class Employee:
    
    def getinfo(self,sal,hours):
        self.sal = sal
        self.hours = hours
    def Addsal(self):
        if self.sal < 5000:
            sal = self.sal + 1000
            print('Input Salary is {} is less than 5000 , Adds 1000 to the salary , Total Salary is : {}'.format(self.sal,sal))
        else:
            print('Salary is : ',self.sal)
    def AddWork(self):
        if hours > 6:
            sal = self.sal + 500
            print('Input Number of hours is {} is more than 6 hours , Adds 500 to the salary , Total salary is : {}'.format(self.hours,sal))
        else:
            print('Salary is : ',self.sal)
        
e = Employee()
sal = int(input('Enter Salary : '))
hours = int(input('Enter hours of work per day : '))
e.getinfo(sal,hours)
e.Addsal()
e.AddWork()

Enter Salary : 10000
Enter hours of work per day : 9
Salary is :  10000
Input Number of hours is 9 is more than 6 hours , Adds 500 to the salary , Total salary is : 10500


**128. Create a class with a function that prints "This is parent class" and its subclass with
another function that prints "This is child class". Now, create an object for each
class and call 1 - function of the parent class by the object of the parent class 2 -
function of the child class by the object of the child class 3 - function of the parent
class by the object of the child class**

In [4]:
class A:
    def a(self):
        print('This is Parent class')
class B(A):
    def b(self):
        print('This is child class')

In [5]:
a = A() # parent class object
b = B() # child class object

In [6]:
a.a() # parent class & parent object

This is Parent class


In [7]:
b.b() # child class and child object

This is child class


In [8]:
b.a() # parent class and child object

This is Parent class


**129. Create a class named 'Member' having the following members: Data members 1 -
Name 2 - Age 3 - Phone number 4 - Address It also has a function named
'printSalary' which prints the salary of the members. Two classes 'Employee' and
'Manager' inherits the 'Member' class. The 'Employee' and 'Manager' classes have
data members 'specialization' and 'department' respectively. Now, assign name,
age, phone number, address and salary to an employee and a manager by making
an object of both of these classes and print the same.**

In [52]:
class Member:
    def __init__(self, name, age, phone_number, address):
        self.name = name
        self.age = age
        self.phone_number = phone_number
        self.address = address

    def printSalary(self):
        pass

class Employee(Member):
    def __init__(self, name, age, phone_number, address, specialization, salary):
        super().__init__(name, age, phone_number, address)
        self.specialization = specialization
        self.salary = salary

    def printSalary(self):
        print("Salary : {}".format(self.salary))

class Manager(Member):
    def __init__(self, name, age, phone_number, address, department, salary):
        super().__init__(name, age, phone_number, address)
        self.department = department
        self.salary = salary

    def printSalary(self):
        print("Salary : {}".format(self.salary))

In [53]:
e = Employee('Ajay',21,9510618897,'Paras Nagar','Data Science',500)

In [54]:
m = Manager('Ajay',21,9510618897,'Paras Nagar','Data Science',5000)

In [56]:
print("Employee Details:")
print("Name:", e.name)
print("Age:", e.age)
print("Phone Number:", e.phone_number)
print("Address:", e.address)
print("Specialization:", e.specialization)
e.printSalary()

print("\nManager Details:")
print("Name:", m.name)
print("Age:", m.age)
print("Phone Number:", m.phone_number)
print("Address:", m.address)
print("Department:", m.department)
m.printSalary()

Employee Details:
Name: Ajay
Age: 21
Phone Number: 9510618897
Address: Paras Nagar
Specialization: Data Science
Salary : 500

Manager Details:
Name: Ajay
Age: 21
Phone Number: 9510618897
Address: Paras Nagar
Department: Data Science
Salary : 5000


**130. Create a class named 'Rectangle' with two data members 'length' and 'breadth' and
two functions to print the area and perimeter of the rectangle respectively. Its
constructor having parameters for length and breadth is used to initialize the length
and breadth of the rectangle. Let class 'Square' inherit the 'Rectangle' class with its
constructor having a parameter for its side (suppose s) calling the constructor of its
parent class. Print the area and perimeter of a rectangle and a square.**

In [59]:
class Rectangle:
    def __init__(self, length, breadth):
        self.length = length
        self.breadth = breadth

    def area(self):
        return self.length * self.breadth

    def perimeter(self):
        return 2 * (self.length + self.breadth)


class Square(Rectangle):
    def __init__(self, side):
        super().__init__(side, side)

r = Rectangle(6, 12)
s = Square(4)

print("Rectangle Details:")
print("Length:", r.length)
print("Breadth:", r.breadth)
print("Area:", r.area())
print("Perimeter:", r.perimeter())

print("\nSquare Details:")
print("Side:", s.length)
print("Area:", s.area())
print("Perimeter:", s.perimeter())

Rectangle Details:
Length: 6
Breadth: 12
Area: 72
Perimeter: 36

Square Details:
Side: 4
Area: 16
Perimeter: 16


**131. Now repeat the above example to print the area of 10 squares. Hint-Use array of
objects**

In [66]:
class Rectangle:
    def __init__(self, length, breadth):
        self.length = length
        self.breadth = breadth

    def area(self):
        return self.length * self.breadth

    def perimeter(self):
        return 2 * (self.length + self.breadth)


class Square(Rectangle):
    def __init__(self, side):
        super().__init__(side, side)

s = [Square(2), Square(3), Square(4), Square(5), Square(6),
           Square(7), Square(8), Square(9), Square(10), Square(11)]

for i, s in enumerate(s, start=1):
    print("\nSquare {} Details:".format(i))
    print("Side:", square.length)
    print("Area:", square.area())
    print("Perimeter:", square.perimeter())


Square 1 Details:
Side: 11
Area: 121
Perimeter: 44

Square 2 Details:
Side: 11
Area: 121
Perimeter: 44

Square 3 Details:
Side: 11
Area: 121
Perimeter: 44

Square 4 Details:
Side: 11
Area: 121
Perimeter: 44

Square 5 Details:
Side: 11
Area: 121
Perimeter: 44

Square 6 Details:
Side: 11
Area: 121
Perimeter: 44

Square 7 Details:
Side: 11
Area: 121
Perimeter: 44

Square 8 Details:
Side: 11
Area: 121
Perimeter: 44

Square 9 Details:
Side: 11
Area: 121
Perimeter: 44

Square 10 Details:
Side: 11
Area: 121
Perimeter: 44


**132. Create a class named 'Shape' with a function to print "This is a shape". Then create
two other classes named 'Rectangle' and 'Circle' inheriting the Shape class, both
having a function to print "This is rectangular shape" and "This is circular shape"
respectively. Create a subclass 'Square' of 'Rectangle' having a function to print
"Square is a rectangle". Now call the function of the 'Shape' and the 'Rectangle'
class by the object of the 'Square' class.**

In [69]:
class Shape:
    def print_shape(self):
        print("This is a shape")

class Rectangle(Shape):
    def print_shape(self):
        print("This is rectangular shape")

class Circle(Shape):
    def print_shape(self):
        print("This is circular shape")

class Square(Rectangle):
    def print_shape(self):
        super().print_shape()
        print("Square is a rectangle")

square = Square()

In [71]:
square.print_shape()

This is rectangular shape
Square is a rectangle


**133. Write a program containing a possible exception. Use a try block to throw it and a
catch block to handle it properly.**

In [76]:
def divide_numbers(a, b):
    try:
        result = a / b
        return result
    except ZeroDivisionError as e:
        print("Error: {}".format(e))
        return None

first = int(input("Enter the first number : "))
second = int(input("Enter the second number : "))

result = divide_numbers(first, second)

if result is not None:
    print("Result of division: {}".format(result))
else:
    print("Cannot divide by zero.")

Enter the first number : 2
Enter the second number : 0
Error: division by zero
Cannot divide by zero.


**134. Write a program that illustrates the application of multiple catch statements.**

In [78]:
def divide_numbers(a, b):
    try:
        result = a / b
        return result
    except ZeroDivisionError as e:
        print("Error: {}. Cannot divide by zero.".format(e))
    except ValueError as e:
        print("Error: {}. Please enter valid numeric values.".format(e))
    except Exception as e:
        print("An unexpected error occurred: {}".format(e))
    finally:
        print("This block always executes, whether an exception occurred or not.")

try:
    first = int(input("Enter the numerator: "))
    second = int(input("Enter the denominator: "))
    result = divide_numbers(first, second)
    if result is not None:
        print(f"Result of division: {result}")
except KeyboardInterrupt:
    print("\nProgram interrupted by the user.")
except Exception as e:
    print("An unexpected error occurred outside the function: {}".format(e))

Enter the numerator: 1
Enter the denominator: 0
Error: division by zero. Cannot divide by zero.
This block always executes, whether an exception occurred or not.


**135. Write a program which uses catch(...) handler.**

In [80]:
def example_function():
    try:
        result = 10 / 0
        print(result)
    except Exception as e:
        print("An exception occurred: {}".format(e))
    finally:
        print("This block always executes, whether an exception occurred or not.")

example_function()

An exception occurred: division by zero
This block always executes, whether an exception occurred or not.


**136. Write a program that demonstrates how certain exception types are not allowed to
be thrown.**

In [81]:
try:
    raise SyntaxError("This is not allowed.")
except SyntaxError as e:
    print(f"Caught a SyntaxError: {e}")
except Exception as e:
    print(f"Caught an exception: {e}")

Caught a SyntaxError: This is not allowed.


**137. Write a program to demonstrate the concept of rethrowing an exception.**

In [83]:
def risky_function():
    try:
        result = 10 / 0
        print(result)
    except ZeroDivisionError as e:
        print("Caught a ZeroDivisionError: {}".format(e))
        raise ValueError("Custom error message") from e

try:
    risky_function()
except ValueError as e:
    print("Caught a ValueError: {}".format(e))
except ZeroDivisionError as e:
    print("This should not be reached due to rethrowing: {}".format(e))
except Exception as e:
    print("Caught a generic exception: {}".format(e))

Caught a ZeroDivisionError: division by zero
Caught a ValueError: Custom error message


**138. Write a program with the following:
(a) A function to read two double type numbers from keyboard
(b) A function to calculate the division of these two numbers
(c) A try block to throw an exception when a wrong type of data is keyed in
(d) A try block to detect and throw an exception if the condition "divide-by-zero"
occurs(e) Appropriate catch block to handle the exceptions thrown**

In [84]:
def read_double():
    try:
        num = float(input("Enter a double type number: "))
        return num
    except ValueError:
        print("Invalid input. Please enter a valid double type number.")
        raise

def calculate_division(num1, num2):
    try:
        result = num1 / num2
        return result
    except ZeroDivisionError:
        print("Error: Cannot divide by zero.")
        raise

try:
    numerator = read_double()
    denominator = read_double()

    result = calculate_division(numerator, denominator)
    print(f"The result of division is: {result}")

except Exception as e:
    print(f"An unexpected error occurred: {e}")

Enter a double type number: 2
Enter a double type number: 7
The result of division is: 0.2857142857142857


**139. Write a main program that calls a deeply nested function containing an exception
Incorporate necessary exception handling mechanism.**

In [85]:
def deep_nested_function():
    try:
        result = 10 / 0
        print(result)
    except ZeroDivisionError:
        print("Error: Cannot divide by zero.")
        raise

def main_program():
    try:
        deep_nested_function()
    except Exception as e:
        print("An unexpected error occurred in the main program: {}".format(e))

main_program()

Error: Cannot divide by zero.
An unexpected error occurred in the main program: division by zero
